api/odigos/v1alpha1/instrumentation_types.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/odigos-io/odigos/k8sutils/pkg/workload"
+)
+
+// Instrumentation configures an application for auto-instrumentation.
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels=metadata.labels.odigos.io/config=1
+// +kubebuilder:metadata:labels=metadata.labels.odigos.io/system-object=true
+// +kubebuilder:printcolumn:name="Workload",type=string,JSONPath=`.spec.workload.name`
+// +kubebuilder:printcolumn:name="Kind",type=string,JSONPath=`.spec.workload.kind`
+// +kubebuilder:printcolumn:name="Namespace",type=string,JSONPath=`.spec.workload.namespace`
+type Instrumentation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   InstrumentationSpec   `json:"spec,omitempty"`
+	Status InstrumentationStatus `json:"status,omitempty"`
+}
+
+type InstrumentationSpec struct {
+	// Workload represents the workload or namespace to be instrumented.
+	// This field is required upon creation and cannot be modified.
+	// +kubebuilder:validation:Required
+	Workload workload.PodWorkload `json:"workload,omitempty"`
```

Comment:
Since it's required, I think we should not have the `omitempty` here.

```suggestion
	Workload workload.PodWorkload `json:"workload"`
```